src/tests/cli_tests.py

```diff
@@ -1867,6 +1867,44 @@ def test_additional_subkeys_reuse_password(self):
         self.assertEqual(ret, 0)
         clear_keyrings()
 
+    def test_set_expire(self):
+        # Primary key with empty password
+        ret, _, _ = run_proc(RNPK, ['--homedir', RNPDIR, '--password=',
+                                    '--userid', 'primary_with_empty_password@rnp', '--generate-key'])
+        self.assertEqual(ret, 0, KEY_GEN_FAILED)
+
+        ret, _, err = run_proc(RNPK, ['--homedir', RNPDIR, '--edit-key', '--set-expire', '-1', 'primary_with_empty_password@rnp'])
+        self.assertNotEqual(ret, 0)
+        self.assertRegex(err, r'Failed to set key expiration.')
```

Comment:
I would recommend to extend test cases with the following checks:
- make sure that expiration didn't change via `--list-packets`
- make sure that expiration is set to exactly same value which was passed via the `--list-packets`.